### ISSUE TITLE
[FR] save a custom namingconvention for the copy&import workflow

### ISSUE THREAD

**Issue by Solarer** [darktable-org/darktable#9613]
**Describe the solution you'd like**
When I copy&import pictures they usually always go into the same parent folder /home/pictures but sometimes I want to import into a different folder /home/projects. I also might want to use a different file naming convention for all my project related pictures. It would be nice if I could save&load multiple presets of the copy&import workflow.

These settings are currently not included in the import module presets.

**Comment by phweyland** [comment]
I would agree on the principle. 

This said, these settings are shared with tethering (that's why they are still in global preferences). To make it work we should have specific settings for tethering. 

And now you have a folder selection (thanks to @johnny-bit)  "based directory naming pattern", which turns easier any change,

**Comment by johnny-bit** [comment]
You seem keen on overusing presets where one might not even think about using presets...

I don't see it as a viable option. For example all my images live in `/home/username/pictures/` under certain schema, that allows me to work on both private an commercial work in same structure. Switching base dir after my recent PR is like 2 clicks. Similar to applying preset.
So no, I don't think there should be presets for copy&import. I feel like it's forcing presets onto something that for majority users ought to be a single global prefference.

**Comment by Solarer** [comment]
> You seem keen on overusing presets where one might not even think about using presets...

You are referring to the the discussion regarding tags ;) I am rarely using presets - I just thought I found a bug in that module and wanted to report it. That was a mistake on my end and it is ok that it was closed. But I do think that having the option to have multiple import paths could be quite useful for some people. If you don't mind I would like to keep it open just in case more people are interested.

**Comment by quovadit** [comment]
I agree with @Solarer , that saving naming-conventions to chose from would be great.
This would be much better than just the new folder-selection, because it's not only about the base-folder, but also about the variables.

**Comment by quovadit** [comment]
This is how I would implement the custom namingconvention:

### 1. Import window:
![image](https://user-images.githubusercontent.com/5677563/134823417-c8b8e609-f600-4508-816e-080bdb67fdd3.png)

Drop Down to chose a 'naming rule'.
The default rule uses yyyymmdd for folder+file but no jobcode, therefor the jobcode field is inactive.
There is a preview for folder- and filename to check, if the naming rule is as expected before copying.

To add custom rules, a click on the gear (or hamburger > preferences) leads to the ...

### 2. Preferences window:
![image](https://user-images.githubusercontent.com/5677563/134823983-64cf522a-c687-49d0-8b7b-ccc46ce3f91f.png)

List of naming rules selectable during import.
You can edit/duplicate existing rules, or create them from scratch ('new').

'Edit' leads to the ...

### 3. Edit naming rule window
![image](https://user-images.githubusercontent.com/5677563/134823655-4e11455e-c8b8-4f89-af3b-fb3b230cdc16.png)

You define the name of the rule and its rules.
The button '+add' opens a drop down to easily add one variable (e.g. $(FILE_NAME) ) at the cursors position
The drop-down 'date type' lets you chose, which kind of date should be used for $(YEAR) etc:
- day of import (current default)
- exif (replaces $(YEAR) ... with $(EXIF_YEAR) ...)
- first exif (uses oldest exif date for all images, relevant if importing images from multiple days)
- custom (= current field 'override today's date')  

.


Any input is extremely welcome!
